engine/runtime/src/main/java/org/enso/interpreter/node/callable/function/BlockNode.java

```diff
@@ -44,4 +48,16 @@ public Object executeGeneric(VirtualFrame frame) {
     }
     return returnExpr.executeGeneric(frame);
   }
+
+  @Override
+  public InstrumentableNode materializeInstrumentableNodes(
+      Set<Class<? extends Tag>> materializedTags) {
+    if (materializedTags.contains(StandardTags.StatementTag.class)) {
+      for (int i = 0; i < statements.length; i++) {
+        statements[i] = insert(new StatementNode(statements[i]));
```

Comment:
Just to be sure, I'd check if `statements[i]` isn't a `StatementNode` already. The docs suggest this method may be called multiple times.

Reply:
Good idea: [4c7025d](https://github.com/enso-org/enso/pull/3432/commits/4c7025d72298a14b8d2c74e1f4f41581d69abda0)